nautobot/docs/additional-features/jobhooks.md

```diff
@@ -0,0 +1,21 @@
+# Jobhooks
+
+A jobhook is a mechanism for automatically starting a job when an object is changed. Jobhooks are similar to [webhooks](../models/extras/webhook.md) except change events initiate a `JobHookReceiver` job instead of a web request. Jobhooks are configured in the web UI under Jobs > Jobhooks.
+
+## Jobhook receivers
+
+Jobhooks are only able to initiate a specific type of job called a jobhook receiver. These are jobs that subclass the `nautobot.extras.jobs.JobHookReceiver` class. Jobhook receivers are similar to normal jobs except they are hard coded to accept only an `object_change` [variable](jobs.md#variables). Jobhook receivers are hidden from the UI by default but are otherwise identical to any other job.
+
+!!! important
+    To prevent negatively impacting system performance through an infinite loop, a change that was made by a jobhook receiver job will not trigger another jobhook receiver job to run.
+
```

Comment:
Maybe add a note like the following, if accurate?

```
!!! note
    For compatibility with the Python API of the base `nautobot.extras.jobs.Job` class, the `JobHookReceiver.run()` method accepts a `commit` parameter in addition to the `data` parameter that contains the `ObjectChange` instance that triggered the JobHook. The `commit` flag will always be set as `True` when a `JobHookReceiver` runs, and so it can be ignored in this case.
```

Alternatively, I'm still liking the idea of providing a default `run()` implementation in JobHookReceiver that calls a separate method that's more specifically targeted toward this use case, something like:

```python
def run(self, data, commit):
   """JobHookReceiver subclasses generally shouldn't need to override this method."""
   object_change = data["object_change"]
   # perhaps also calculate the diffs that this object_change represents, as mentioned in my earlier comments on this PR?
   self.receive_jobhook(change=object_change, action=object_change.action, changed_object=object_change.changed_object, ...)

def receive_jobhook(self, change, action, changed_object, ...):
    """Method to be implemented by concrete JobHookReceiver subclasses."""
    raise NotImplementedError
```